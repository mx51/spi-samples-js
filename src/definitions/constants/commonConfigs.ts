import { SpiFlow, SpiStatus } from '@mx51/spi-client-js';
// configuration types
export const TEXT_FORM_CONFIGURATION_AUTO_ADDRESS_VALUE = 'auto';
export const TEXT_FORM_CONFIGURATION_EFTPOS_ADDRESS_VALUE = 'eftpos';
// payment provider codes
export const TEXT_FORM_DEFAULT_VALUE = 'Please type or select SPI';
export const TEXT_FORM_MODAL_CODE_TILL = 'till';
export const TEXT_FORM_MODAL_CODE_WESTPAC = 'wbc';
// form validations
export const TEXT_FORM_VALIDATION_API_KEY_TEXTFIELD = 'Please enter an API Key.';
export const TEXT_FORM_VALIDATION_POS_ID_TEXTFIELD = 'Please enter a POS ID.';
export const TEXT_FORM_VALIDATION_PROVIDER_TEXTFIELD = 'Please enter a supported provider.';
export const TEXT_FORM_VALIDATION_SERIAL_NUMBER_TEXTFIELD = 'Please enter a 9 digits Serial number.';
export const TEXT_FORM_VALIDATION_EFTPOS_ADDRESS_TEXTFIELD = 'Please enter a valid EFTPOS address.';
// pair flow/status/state
export const SPI_PAIR_FLOW = {
  Idle: SpiFlow.Idle,
  Pairing: SpiFlow.Pairing,
  Transaction: SpiFlow.Transaction,
};
export const SPI_PAIR_STATUS = {
  PairedConnected: SpiStatus.PairedConnected,
  PairedConnecting: SpiStatus.PairedConnecting,
  Unpaired: SpiStatus.Unpaired,
};
// pair flow drawer width 4/12 (based on design required)
export const drawerWidth = 33.33333;
// spi service constants
export const spiEvents = {
  spiAutoAddressResolutionFailed: 'AutoAddressResolutionFailed',
  spiBatteryLevelChanged: 'BatteryLevelChanged',
  spiDeviceAddressChanged: 'DeviceAddressChanged',
  spiSecretsChanged: 'SecretsChanged',
  spiFlowChanged: 'spiFlowChanged',
  spiPairingFlowStateChanged: 'PairingFlowStateChanged',
  spiPong: 'SpiPong',
  spiStatusChanged: 'StatusChanged',
  spiTerminalSetupsChanged: 'TerminalConfigChanged',
  spiTerminalStatusChanged: 'TerminalStatusChanged',
  spiTxFlowStateChanged: 'TxFlowStateChanged',
  spiTxUpdateMessage: 'TransactionUpdateMessage',
  TerminalStatusResponse: 'terminal_status',
};

export const commonPairErrorMessage = 'Unable to pair. Please check your configuration details and try again.';
