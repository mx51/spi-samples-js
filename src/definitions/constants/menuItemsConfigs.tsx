import React from 'react';
// paths & links constants
import {
  LINK_FLEET_SETTINGS,
  LINK_PAIR,
  LINK_PAY_AT_TABLE,
  LINK_PRE_AUTH,
  LINK_PURCHASE,
  LINK_REFUND,
  LINK_TERMINALS,
  PATH_FLEET_SETTINGS,
  PATH_PAIR,
  PATH_PAY_AT_TABLE,
  PATH_PRE_AUTH,
  PATH_PURCHASE,
  PATH_REFUND,
  PATH_TERMINALS,
} from './routerConfigs';
// icons
import { ReactComponent as IconFleetSettings } from '../../images/FleetSettingsIcon.svg';
import { ReactComponent as IconPairing } from '../../images/PairingIcon.svg';
import { ReactComponent as IconPayAtTable } from '../../images/PayAtTableIcon.svg';
import { ReactComponent as IconPreAuthorisation } from '../../images/PreAuthorisationIcon.svg';
import { ReactComponent as IconPurchase } from '../../images/PurchaseIcon.svg';
import { ReactComponent as IconRefund } from '../../images/RefundIcon.svg';
import { ReactComponent as IconTerminalsList } from '../../images/TerminalsListIcon.svg';

export default {
  samplePos: [
    {
      name: LINK_PURCHASE,
      path: PATH_PURCHASE,
      icon: <IconPurchase />,
    },
    {
      name: LINK_REFUND,
      path: PATH_REFUND,
      icon: <IconRefund />,
    },
    {
      name: LINK_PAY_AT_TABLE,
      path: PATH_PAY_AT_TABLE,
      icon: <IconPayAtTable />,
    },
    {
      name: LINK_PRE_AUTH,
      path: PATH_PRE_AUTH,
      icon: <IconPreAuthorisation />,
    },
  ],
  terminals: [
    {
      name: LINK_TERMINALS,
      path: PATH_TERMINALS,
      icon: <IconTerminalsList />,
    },
    {
      name: LINK_PAIR,
      path: PATH_PAIR,
      icon: <IconPairing />,
    },
    {
      name: LINK_FLEET_SETTINGS,
      path: PATH_FLEET_SETTINGS,
      icon: <IconFleetSettings />,
    },
  ],
};
