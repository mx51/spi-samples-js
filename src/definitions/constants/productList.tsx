export default [
  {
    name: 'Mocha',
    price: 4.2,
    image: 'images/product/mocha.png',
  },
  {
    name: 'Honey Latte',
    price: 4.8,
    image: 'images/product/honeyLatte.png',
  },
  {
    name: 'Flat White',
    price: 3.8,
    image: 'images/product/flatWhite.png',
  },
  {
    name: 'Chai Latte',
    price: 4.5,
    image: 'images/product/chaiLatte.png',
  },
  {
    name: 'Rose Tea Latte',
    price: 4.5,
    image: 'images/product/roseTeaLatte.png',
  },
  {
    name: 'Cappuccino',
    price: 3.8,
    image: 'images/product/cap.png',
  },
  {
    name: 'Latte',
    price: 3.8,
    image: 'images/product/latte.png',
  },
  {
    name: 'Production Test Item',
    price: 0.01,
    image: 'images/product/testItem.png',
  },
];
