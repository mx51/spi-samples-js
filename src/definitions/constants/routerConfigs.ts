// paths
export const PATH_FLEET_SETTINGS = '/fleet-settings';
export const PATH_PAIR = '/pair';
export const PATH_PAY_AT_TABLE = '/pay-at-table';
export const PATH_PRE_AUTH = '/pre-auth';
export const PATH_PURCHASE = '/';
export const PATH_REFUND = '/refund';
export const PATH_TERMINALS = '/terminals';
// link names
export const LINK_FLEET_SETTINGS = 'Fleet settings';
export const LINK_PAIR = 'Pairing';
export const LINK_PAY_AT_TABLE = 'Pay at Table';
export const LINK_PRE_AUTH = 'Preauthorisation';
export const LINK_PURCHASE = 'Purchase';
export const LINK_REFUND = 'Refund';
export const LINK_TERMINALS = 'Terminals List';
