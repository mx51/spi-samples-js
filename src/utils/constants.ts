import { version } from '../../package.json';

export const posVersion = version;

export const localStorageKeys = {
  receiptConfig: 'Receipt Config',
  patConfig: 'PAT Config',
};
