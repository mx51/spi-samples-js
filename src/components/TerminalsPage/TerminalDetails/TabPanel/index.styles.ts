import { createStyles, makeStyles, Theme } from '@material-ui/core';
import { drawerWidth } from '../../../../definitions/constants/commonConfigs';
import { IFlowPanel } from '../../../FlowPanel/interfaces';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      width: '100%',
      display: 'flex',
      '& .MuiTabs-indicator': {
        backgroundColor: theme.palette.primary.main,
      },
    },
    panel: {
      flexGrow: 1,
      padding: theme.spacing(6),
      transition: theme.transitions.create('margin', {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
      }),
      marginRight: `${-drawerWidth}%`,
    },
    panelShift: {
      transition: theme.transitions.create('margin', {
        easing: theme.transitions.easing.easeOut,
        duration: theme.transitions.duration.enteringScreen,
      }),
      marginRight: 0,
    },
    tabPanelContainer: {
      display: 'inherit',
      width: '100%',
    },
    [theme.breakpoints.between(960, 1025)]: {
      root: {
        '&>main': {
          padding: theme.spacing(6, 2),
        },
        '& form': {
          marginBottom: (showFlowPanel: IFlowPanel) => (showFlowPanel ? 0 : -theme.spacing(2)),
        },
      },
    },
    [theme.breakpoints.between(600, 959)]: {
      root: {
        '&>main': {
          padding: theme.spacing(6, 0),
        },
      },
      pairStatusContainer: {
        flexBasis: (showFlowPanel: IFlowPanel) => (showFlowPanel ? '100%' : '33.333333%'),
        maxWidth: (showFlowPanel: IFlowPanel) => (showFlowPanel ? '100%' : '33.333333%'),
      },
    },
    [theme.breakpoints.down(600)]: {
      root: {
        '&>main': {
          padding: theme.spacing(6, 0),
        },
        '& form': {
          marginBottom: theme.spacing(2),
        },
      },
      tabPanelContainer: {
        flexBasis: '100%',
        width: '100%',
        maxWidth: '100%',
        '&>div': {
          paddingRight: 0,
        },
      },
    },
    // tab showFlowPanel stylings
    text: {
      fontWeight: 300,
    },
    flowButton: {
      color: 'inherit',
      background: 'transparent',
      border: 'none',
      cursor: 'pointer',
    },
    flowIcon: {
      cursor: 'pointer',
    },
    // receipt panel stylings
    detailsPanelContainer: {
      width: '100%',
      display: 'flex',
    },
    detailsPanel: {
      backgroundColor: theme.palette.common.white,
      border: `1px solid ${theme.palette.secondary.main}`,
      borderRadius: theme.spacing(0.5),
      boxSizing: 'border-box',
      flexGrow: 1,
      marginRight: `${-drawerWidth}%`,
      padding: theme.spacing(3),
      transition: theme.transitions.create('margin', {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
      }),
    },
    detailsPanelShift: {
      transition: theme.transitions.create('margin', {
        easing: theme.transitions.easing.easeOut,
        duration: theme.transitions.duration.enteringScreen,
      }),
      marginRight: 0,
    },
    receiptPanel: {
      flexShrink: 0,
      marginLeft: theme.spacing(4),
      width: `${drawerWidth}%`,
    },
    receiptPanelOpened: {
      backgroundColor: theme.palette.common.white,
      border: `1px solid ${theme.palette.secondary.main}`,
      borderRadius: theme.spacing(0.5),
      boxSizing: 'border-box',
      flexShrink: 0,
      marginLeft: theme.spacing(4),
      width: `${drawerWidth}%`,
    },
    preContent: {
      whiteSpace: 'pre-wrap',
    },
  })
);

export default useStyles;
