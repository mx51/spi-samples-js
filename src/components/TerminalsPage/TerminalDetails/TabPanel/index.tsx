import React from 'react';
import { Container } from '@material-ui/core';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import { TxFlowState } from '../../../../definitions/constants/terminalConfigs';
import { useAppSelector } from '../../../../redux/hooks';
import { selectedShowFlowPanel } from '../../../../redux/reducers/CommonSlice/commonSliceSelectors';
import {
  terminalTxFlowFinishedTracker,
  terminalTxFlowReceipt,
  terminalTxFlowReceiptContent,
  terminalTxFlowSuccessTracker,
} from '../../../../redux/reducers/TerminalSlice/terminalsSliceSelectors';
import FlowPanel from '../../../FlowPanel';
import PairFlow from '../../../FlowPanel/PairFlow';
import ReceiptPanel from '../../../ReceiptPanel';
import { TabPanelProps } from '../interfaces';
import useStyles from './index.styles';

export default function TabPanel({
  children,
  index,
  subtitle,
  title,
  value,
  receiptToggle,
  terminal,
}: TabPanelProps): React.ReactElement {
  const showFlowPanel = useAppSelector(selectedShowFlowPanel);
  const classes = useStyles({ showFlowPanel });

  const txFlowReceiptContent = useAppSelector(terminalTxFlowReceiptContent(terminal?.id as string));
  const txFlowReceipt = useAppSelector(terminalTxFlowReceipt(terminal?.id as string));
  const txFlowFinishedState = useAppSelector(terminalTxFlowFinishedTracker(terminal?.id as string));
  const txFlowSuccessState = useAppSelector(terminalTxFlowSuccessTracker(terminal?.id as string));

  const ReceiptContentDOM = () => (
    <>
      {!txFlowFinishedState && txFlowSuccessState === TxFlowState.Unknown && (
        <pre className={classes.preContent}>Settle Initiated. Will be updated with Progress.</pre>
      )}
      {(txFlowSuccessState === TxFlowState.Failed || txFlowReceipt?.errorReason) && txFlowFinishedState && (
        <pre className={classes.preContent}>
          Could not initiate settle: <strong>{txFlowReceipt?.errorReason}</strong>. Please Retry.
        </pre>
      )}
      {txFlowFinishedState && txFlowSuccessState === TxFlowState.Success && txFlowReceiptContent && (
        <pre className={classes.preContent}>{txFlowReceiptContent}</pre>
      )}
    </>
  );

  return (
    <>
      {value === index && (
        <div className={classes.root}>
          <div className={showFlowPanel ? `${classes.panel} ${classes.panelShift}` : classes.panel}>
            <Container maxWidth="lg">
              <Grid className={classes.detailsPanelContainer}>
                <div
                  className={
                    receiptToggle?.settlement || receiptToggle?.settlementEnquiry
                      ? `${classes.detailsPanel} ${classes.detailsPanelShift}`
                      : classes.detailsPanel
                  }
                >
                  <Grid container>
                    <Grid item className={classes.tabPanelContainer}>
                      <Grid container direction="column" justifyContent="space-between" spacing={1}>
                        <Typography variant="h6" component="h1">
                          {title}
                        </Typography>
                        <Typography className={classes.text}>{subtitle}</Typography>
                      </Grid>
                    </Grid>
                  </Grid>
                  <Grid>{children}</Grid>
                </div>
                <Grid
                  className={
                    receiptToggle?.settlement || receiptToggle?.settlementEnquiry
                      ? classes.receiptPanelOpened
                      : classes.receiptPanel
                  }
                >
                  {receiptToggle?.settlement && (
                    <ReceiptPanel title="Settlement Receipt" css={classes.receiptBoxWrapper}>
                      <ReceiptContentDOM />
                    </ReceiptPanel>
                  )}
                  {receiptToggle?.settlementEnquiry && (
                    <ReceiptPanel title="Settlement Enquiry Receipt" css={classes.receiptBoxWrapper}>
                      <ReceiptContentDOM />
                    </ReceiptPanel>
                  )}
                </Grid>
              </Grid>
            </Container>
          </div>
          <FlowPanel>
            <PairFlow terminal={terminal} />
          </FlowPanel>
        </div>
      )}
    </>
  );
}
