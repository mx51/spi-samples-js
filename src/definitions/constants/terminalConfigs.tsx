import { ITerminalConfigurationConstants } from '../../components/TerminalsPage/TerminalDetails/interfaces';
import { ITerminalProps } from '../../redux/reducers/TerminalSlice/interfaces';

export const terminalConfigurationsPartOne = (terminal: ITerminalProps): Array<ITerminalConfigurationConstants> => [
  {
    title: 'Merchant ID',
    content: `${terminal?.merchantId}`,
  },
  {
    title: 'Terminal ID',
    content: `${terminal?.terminalId}`,
  },
  {
    title: 'Serial number',
    content: `${terminal?.serialNumber}`,
  },
  {
    title: 'Battery',
    content: `${terminal?.batteryLevel}${terminal?.batteryLevel && terminal?.batteryLevel !== '-' ? '%' : ''}`,
  },
];

export const terminalConfigurationsPartTwo = (terminal: ITerminalProps): Array<ITerminalConfigurationConstants> => [
  {
    title: 'Payment provider',
    content: `${terminal?.acquirerCode}`,
  },
  {
    title: 'Configuration option',
    content: `${terminal?.autoAddress ? 'Yes' : 'No'}`,
  },
  {
    title: 'EFTPOS address',
    content: `${terminal?.deviceAddress}`,
  },
  {
    title: 'POS ID',
    content: `${terminal?.posId}`,
  },
];

export const payAtTableOptions = [
  {
    name: 'payAtTable',
    title: 'Pay at Table',
    subtitle: 'Enable Pay at Table for this terminal',
  },
  {
    name: 'splitBillEvenly',
    title: 'Split bill evenly',
    subtitle: 'Allow customers to split the bill equally',
  },
  {
    name: 'tipping',
    title: 'Tipping',
    subtitle: 'Allow customers to tip during a Pay at Table transaction',
  },
  {
    name: 'acceptCash',
    title: 'Accept cash',
    subtitle: 'Allow customers to pay by cash during a Pay at Table transaction',
  },
  {
    name: 'retrieveAllTables',
    title: 'Retrieve all tables',
    subtitle: 'Request a list of all open tables and sales from the POS',
  },
  {
    name: 'requireOperatorID',
    title: 'Require Operator ID',
    subtitle: 'Prompt for Operator ID in Pay at Table mode',
  },
  {
    name: 'transactionSummaryReport',
    title: 'Transaction summary report',
    subtitle: 'Print a summary at the end of a Pay at Table transaction',
  },
];

export const payAtTableFields = [
  {
    helperText: 'Customise the ‘Pay at Table’ button displayed on the terminal',
    id: 'payAtTableLabelField',
    label: 'Pay at Table label',
  },
  {
    helperText: 'Customise the ‘Operator ID’ field label displayed on the terminal',
    id: 'operatorIDField',
    label: 'Operator ID label',
  },
  {
    helperText: 'Customise the ‘Table’ field label displayed on the terminal',
    id: 'tableLabelField',
    label: 'Table label',
  },
];
