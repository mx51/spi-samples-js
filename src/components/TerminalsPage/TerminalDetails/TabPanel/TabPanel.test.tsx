import React from 'react';
import { cleanup } from '@testing-library/react';
import TabPanel from '.';
import { TxFlowState } from '../../../../definitions/constants/terminalConfigs';
import mockWithRedux, {
  defaultMockPairFormParams,
  defaultMockTerminals,
  mockReceiptResponse,
  mockTerminalInstanceId,
  pairedMockTerminals,
} from '../../../../utils/tests/common';

describe('Test <TabPanel />', () => {
  afterEach(cleanup);

  test('should match TabPanel snapshot test', () => {
    // Arrange
    const customizedStore = {
      getState: () => ({
        common: { showFlowPanel: true },
        pairForm: defaultMockPairFormParams,
        products: [],
        terminals: defaultMockTerminals,
      }),
      subscribe: jest.fn(),
      dispatch: jest.fn(),
    };
    const container = mockWithRedux(
      <TabPanel
        index={0}
        subtitle="Test tab 1 subtitle"
        title="Test tab 1 title"
        value={0}
        receiptToggle={{
          settlement: true,
          settlementEnquiry: false,
        }}
        terminal={defaultMockTerminals[mockTerminalInstanceId]}
      >
        <div>Test Panel One</div>
      </TabPanel>,
      customizedStore
    );

    // Assert
    expect(container).toMatchSnapshot();
    expect(container.innerHTML.includes('<div>Test Panel One</div>')).toBeTruthy();
    expect(container.innerHTML.includes('Flow')).toBeTruthy(); // proves flow is opened as flow content is part of DOM
  });

  test('should show receipt panel when receiptToggle value is true', () => {
    const container = mockWithRedux(
      <TabPanel
        index={1}
        subtitle="Test tab 2 subtitle"
        title="Test tab 2 title"
        value={1}
        receiptToggle={{
          settlement: false,
          settlementEnquiry: true,
        }}
        terminal={defaultMockTerminals[mockTerminalInstanceId]}
      >
        <div>Test Panel Two</div>
      </TabPanel>
    );

    expect(container.innerHTML.includes('Settlement Enquiry Receipt')).toBeTruthy();
    expect(container.innerHTML.includes('Test Panel Two')).toBeTruthy();
  });

  test('should display receipt loading status message when txFlow finished is false and success is Unknown', () => {
    // Arrange
    const customizedStore = {
      getState: () => ({
        common: { showFlowPanel: false },
        pairForm: defaultMockPairFormParams,
        terminals: {
          ...defaultMockTerminals,
          [mockTerminalInstanceId]: {
            ...defaultMockTerminals[mockTerminalInstanceId],
            receipt: mockReceiptResponse,
            txFlow: {
              finished: false,
              success: TxFlowState.Unknown,
            },
          },
        },
      }),
      subscribe: jest.fn(),
      dispatch: jest.fn(),
    };
    const newMockContainer = mockWithRedux(
      <TabPanel
        index={0}
        subtitle="Test tab 1 subtitle"
        title="Test tab 1 title"
        value={0}
        receiptToggle={{
          settlement: false,
          settlementEnquiry: true,
        }}
        terminal={pairedMockTerminals[mockTerminalInstanceId]}
      >
        <div>Test Panel One</div>
      </TabPanel>,
      customizedStore
    );

    // Assert
    expect(newMockContainer.innerHTML.includes('Settle Initiated. Will be updated with Progress.')).toBe(true);
  });

  test('should display receipt with error when txFlow finished is false and success is Failed', () => {
    // Arrange
    const customizedStore = {
      getState: () => ({
        common: { showFlowPanel: false },
        pairForm: defaultMockPairFormParams,
        terminals: {
          ...defaultMockTerminals,
          [mockTerminalInstanceId]: {
            ...defaultMockTerminals[mockTerminalInstanceId],
            receipt: mockReceiptResponse,
            txFlow: {
              finished: true,
              success: TxFlowState.Failed,
            },
          },
        },
      }),
      subscribe: jest.fn(),
      dispatch: jest.fn(),
    };
    const newMockContainer = mockWithRedux(
      <TabPanel
        index={0}
        subtitle="Test tab 1 subtitle"
        title="Test tab 1 title"
        value={0}
        receiptToggle={{
          settlement: false,
          settlementEnquiry: true,
        }}
        terminal={pairedMockTerminals[mockTerminalInstanceId]}
      >
        <div>Test Panel One</div>
      </TabPanel>,
      customizedStore
    );

    // Assert
    expect(newMockContainer.innerHTML.includes('Settle Initiated. Will be updated with Progress.')).toBe(false);
    expect(newMockContainer.innerHTML.includes('Could not initiate settle:')).toBe(true);
  });

  test('should display receipt content when txFlow finished is true and success is Success', () => {
    // Arrange
    const customizedStore = {
      getState: () => ({
        common: { showFlowPanel: false },
        pairForm: defaultMockPairFormParams,
        terminals: {
          ...defaultMockTerminals,
          [mockTerminalInstanceId]: {
            ...defaultMockTerminals[mockTerminalInstanceId],
            receipt: mockReceiptResponse,
            txFlow: {
              finished: true,
              success: TxFlowState.Success,
            },
          },
        },
      }),
      subscribe: jest.fn(),
      dispatch: jest.fn(),
    };
    const newMockContainer = mockWithRedux(
      <TabPanel
        index={0}
        subtitle="Test tab 1 subtitle"
        title="Test tab 1 title"
        value={0}
        receiptToggle={{
          settlement: false,
          settlementEnquiry: true,
        }}
        terminal={pairedMockTerminals[mockTerminalInstanceId]}
      >
        <div>Test Panel One</div>
      </TabPanel>,
      customizedStore
    );

    // Assert
    expect(newMockContainer.innerHTML.includes('EFTPOS FROM TEST TENANT')).toBe(true);
  });
});
