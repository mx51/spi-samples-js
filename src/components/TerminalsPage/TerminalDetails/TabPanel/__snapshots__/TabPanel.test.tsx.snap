// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <TabPanel /> should match TabPanel snapshot test 1`] = `
<div>
  <div
    class="makeStyles-root-1 makeStyles-root-16"
  >
    <div
      class="makeStyles-panel-2 makeStyles-panelShift-3"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <div
          class="MuiGrid-root makeStyles-detailsPanelContainer-9"
        >
          <div
            class="makeStyles-detailsPanel-10 makeStyles-detailsPanelShift-11"
          >
            <div
              class="MuiGrid-root MuiGrid-container"
            >
              <div
                class="MuiGrid-root makeStyles-tabPanelContainer-4 MuiGrid-item"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column MuiGrid-justify-content-xs-space-between"
                >
                  <h1
                    class="MuiTypography-root MuiTypography-h6"
                  >
                    Test tab 1 title
                  </h1>
                  <p
                    class="MuiTypography-root makeStyles-text-6 MuiTypography-body1"
                  >
                    Test tab 1 subtitle
                  </p>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root"
            >
              <div>
                Test Panel One
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root makeStyles-CustomContentPanelOpened-13"
          >
            <div
              class="MuiGrid-root makeStyles-receiptBoxWrapper-15"
            >
              <div
                class="MuiBox-root MuiBox-root-23 makeStyles-box-22"
              >
                <div
                  class="MuiGrid-root makeStyles-gridContainer-18"
                >
                  <div
                    class="MuiGrid-root makeStyles-receiptGrid-19 MuiGrid-item MuiGrid-grid-md-12"
                  >
                    <h1
                      class="MuiTypography-root MuiTypography-h6"
                    >
                      Settlement Receipt
                    </h1>
                  </div>
                  <div
                    class="MuiGrid-root makeStyles-buttonGrid-20 MuiGrid-item MuiGrid-grid-md-12"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-24"
      data-test-id="flowPanel"
    >
      <div
        class="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-25 MuiDrawer-paperAnchorRight MuiDrawer-paperAnchorDockedRight MuiPaper-elevation0"
      >
        <div
          class="MuiBox-root MuiBox-root-29 makeStyles-flowBoxWrapper-26"
        >
          <div
            class="MuiBox-root MuiBox-root-30 makeStyles-flowBox-27"
          >
            <h1
              class="MuiTypography-root MuiTypography-h6"
            >
              Flow
            </h1>
            <pre
              class="makeStyles-flowContent-28"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
